### DISCUSSION TITLE
Get device' config context in custom Script

### DISCUSSION THREAD

**Discussion by jnc27** [netbox-community/netbox#13251] [Q&A]
Hi,
I would like to retrieve the rendered config context of a device. Apparently there is no "config_context" in Device model. I only see local_context_data in API documentation.
I've seen that there is the DeviceWithConfigContext model. But I don't know how to use it? tried to import it in dcim.models, but that didn't work.
Thank you.

**Comment by abhi1693** [comment]
If you are using netbox 3.5+

There are a few steps you'll have to do to get this via orm

```
 context_data = device.get_config_context()
 context_data.update({'device': device})
 config_template = device.get_config_template()
 config_template.render(context_data)
```

Note, you'll have to add appropriate checks to ensure the code does not crash
